SDWebImage/Core/SDImageIOAnimatedCoder.m

```diff
@@ -895,6 +895,21 @@ - (NSData *)encodedDataWithFrames:(NSArray<SDImageFrame *> *)frames loopCount:(N
         maxPixelSize = maxPixelSizeValue.CGSizeValue;
 #endif
     }
+    // HDR Encoding
+    NSUInteger encodeToHDR = 0;
+    if (options[SDImageCoderEncodeToHDR]) {
+        encodeToHDR = [options[SDImageCoderEncodeToHDR] unsignedIntegerValue];
+    }
+    if (@available(macOS 15, iOS 18, tvOS 18, watchOS 11, *)) {
+        if (encodeToHDR == 1) {
+            properties[(__bridge NSString *)kCGImageDestinationEncodeRequest] = (__bridge NSString *)kCGImageDestinationEncodeToISOHDR;
```

Comment:
This API need Xcode 16 SDK
Should we still support Xcode 15 SDK ? Copy the raw key string into our code instead for backward compatibility ?

Reply:
Done. Still support Xcode 15 to build SDWebImage's binary